api/src/client/DynamicsLicenseStatusClient.ts

```diff
@@ -0,0 +1,64 @@
+import { LogWriterType } from "@libs/logWriter";
+import { LicenseSearchNameAndAddress, LicenseStatusResult } from "@shared/license";
+import { searchBusinessAddressesForMatch } from "../domain/license-status/searchBusinessAddressesForMatch";
+import {
+  AccessTokenClient,
+  BusinessAddressClient,
+  BusinessIdClient,
+  ChecklistItemsClient,
+  LicenseApplicationIdClient,
+  SearchLicenseStatus,
+} from "../domain/types";
+
+type Config = {
+  tokenFactory: AccessTokenClient;
+  businessIdFactory: BusinessIdClient;
+  businessAddressFactory: BusinessAddressClient;
+  applicationIdFactory: LicenseApplicationIdClient;
+  checkListItemFactory: ChecklistItemsClient;
+};
+
+export const DynamicsLicenseStatusClient = (
+  logWriter: LogWriterType,
+  config: Config
+): SearchLicenseStatus => {
+  return async (
+    nameAndAddress: LicenseSearchNameAndAddress,
+    licenseType: string
+  ): Promise<LicenseStatusResult> => {
+    const accessToken = await config.tokenFactory.getAccessToken();
+
+    const businessIds = await config.businessIdFactory.getBusinessId(accessToken, nameAndAddress.name);
+
+    const mappedBusinessAddressesByIds = await config.businessAddressFactory.getMappedBusinessAddressesByIds(
+      accessToken,
+      businessIds
+    );
+
+    const matchedBusinessId = searchBusinessAddressesForMatch(mappedBusinessAddressesByIds, nameAndAddress);
+
+    const applicationIdResponse = await config.applicationIdFactory.getLicenseApplicationId(
+      accessToken,
+      matchedBusinessId,
+      licenseType
+    );
+
+    const checkListItems = await config.checkListItemFactory.getChecklistItems(
+      accessToken,
+      applicationIdResponse.applicationId
+    );
+
+    const response = {
+      status: applicationIdResponse.licenseStatus,
+      expirationISO: applicationIdResponse.expirationDate,
+      checklistItems: [...checkListItems],
+    };
+
+    const logId = logWriter.GetId();
+    logWriter.LogInfo(
+      `Dynamics License Status Client - Id:${logId} - Response:  ${JSON.stringify(response)}`
+    );
+
+    return response;
```

Comment:
[boulder] This is good but I think it's missing error handling - both in the code and tests. What happens if any of these awaited calls throw an error?

There's no need to get specific with different types of errors, but I think there should be a test for what this returns if any one of the many calls goes wrong.

Reply:
added both instances of errors (status code error and NO_MATCH)